tools/state-viewer/src/cli.rs

```diff
@@ -160,11 +160,21 @@ pub struct ApplyCmd {
     height: BlockHeight,
     #[clap(long)]
     shard_id: ShardId,
+    #[clap(long)]
+    use_flat_storage: bool,
```

Comment:
Are there any preconditions for this to work? Such as that flat storage is at the `height` argument from above (+/- deltas)? 

Can you codify those conditions and fail early if any are not met? This is just a suggestions given this is a debugging tool.

Reply:
I agree that this CLI is sub-optimal and is 'leaking an abstraction'.
Flat storage can only be used for processing one of the 3 most recent blocks.
The `view-state apply` commands can be involved for any block.

In general using flat storage when it's available has no downsides, it gives you the correct function call results and the correct gas usage. The only downside is that it's not available for most blocks.

An option to apply a block without FS lets you get correct function call results regardless (but not the correct gas usage).

Last quarter there was a plan to do something about making it possible to replay anything using FS, but it was deprioritized. This sub-optimal CLI is a short-term fix that lets us replay the recent blocks correctly.

> codify

Not convinced about it. At the moment I want an extra knob for extra control. The project mentioned above should take care of it eventually.